src/c2goto/cprover_library.cpp

```diff
@@ -149,79 +127,36 @@ void add_cprover_library(contextt &context, const messaget &message_handler)
   goto_functionst goto_functions;
   std::multimap<irep_idt, irep_idt> symbol_deps;
   std::list<irep_idt> to_include;
-  char symname_buffer[288];
-  FILE *f;
-  uint8_t **this_clib_ptrs;
-  uint64_t size;
-  int fd;
+  const buffer *clib;
 
-  if(config.ansi_c.word_size == 32)
+  switch(config.ansi_c.word_size)
   {
-    if(config.ansi_c.use_fixed_for_float)
-    {
-      this_clib_ptrs = &clib_ptrs[0][0];
-    }
-    else
-    {
-      this_clib_ptrs = &clib_ptrs[2][0];
-    }
-  }
-  else if(config.ansi_c.word_size == 64)
-  {
-    if(config.ansi_c.use_fixed_for_float)
-    {
-      this_clib_ptrs = &clib_ptrs[1][0];
-    }
-    else
-    {
-      this_clib_ptrs = &clib_ptrs[3][0];
-    }
-  }
-  else
-  {
-    if(config.ansi_c.word_size == 16)
-    {
-      message_handler.warning(
-        "Warning: this version of ESBMC does not have a C library "
-        "for 16 bit machines");
-      return;
-    }
+  case 16:
+    message_handler.warning(
+      "Warning: this version of ESBMC does not have a C library "
+      "for 16 bit machines");
+    return;
```

Comment:
I wonder if we should keep these checks for 16-bit, do we have any user on it on your side @lucasccordeiro? 

I mean, we don't even have models for 16-bit arch.

Reply:
> I wonder if we should keep these checks for 16-bit, do we have any user on it on your side @lucasccordeiro?

I don't think so.

@rafaelsamenezes: do you have any thoughts here?

Reply:
We can open a new issue about removing them.